tests/apollo/util/bft.py

```diff
@@ -1341,13 +1341,15 @@ async def check_initital_key_exchange(self, stop_replicas=True):
                 self.stop_all_replicas()
             return lastExecutedVal
 
-    async def assert_successful_pre_executions_count(self, replica_id, num_requests):
+    async def assert_successful_pre_executions_count(self, replica_id, num_requests, threshold=1.0, timeout_in_sec=10):
         try:
+            threshold = 1.0 if(threshold > 1.0) else threshold
             pre_proc_req = 0
             total_pre_exec_requests_executed = 0
-            with trio.fail_after(10):
-                while pre_proc_req < num_requests or \
-                        total_pre_exec_requests_executed < num_requests:
+            threshold_request = num_requests * threshold;
+            with trio.fail_after(timeout_in_sec):
+                while pre_proc_req < threshold_request or \
+                        total_pre_exec_requests_executed < threshold_request:
                     key1 = ["preProcessor", "Counters", "preProcReqCompleted"]
                     pre_proc_req = await self.metrics.get(replica_id, *key1) - self.initial_preexec_sent
```

Comment:
if possible, please revert the change in the shared utilities code. If you really need this change, I'd suggest to have it carefully reviewed by @KTeofanidis ...

Reply:
This change will not be needed as we are reducing the number of tests.